src/v/cluster/topic_table.cc

```diff
@@ -670,23 +671,20 @@ void topic_table::notify_waiters() {
                                         - _pending_deltas.begin();
 
     /// Consume all pending deltas
-    if (!_waiters.empty()) {
-        std::vector<std::unique_ptr<waiter>> active_waiters;
-        active_waiters.swap(_waiters);
-        for (auto& w :
-             std::span{active_waiters.begin(), active_waiters.size() - 1}) {
-            w->promise.set_value(_pending_deltas);
-        }
+    if (_waiter) {
+        std::unique_ptr<waiter> active_waiter;
+        active_waiter.swap(_waiter);
+        active_waiter->promise.set_value(std::move(_pending_deltas));
 
-        active_waiters[active_waiters.size() - 1]->promise.set_value(
-          std::move(_pending_deltas));
         std::exchange(_pending_deltas, {});
```

Comment:
Maybe this line should be rolled into the one above, like `promise.set_value(std::exchange(_pending_deltas, {}))`?

Reply:
Nice catch, switching to this now.